Client/Frontend/Browser/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public protocol RecentlyClosedTabs {
+    func addTab(url: NSURL, title: String?, faviconURL: String?)
+    func getTabs() -> [ClosedTab]
+    func clearTabs()
+    func count() -> Int
+}
+
+public class RecentlyClosedTabsStore: RecentlyClosedTabs {
+
+    public init() {
+    }
+
+    public func addTab(url: NSURL, title: String?, faviconURL: String?) {
+        if let recentlyClosedTab = ClosedTab(url: url, title: title ?? "", faviconURL: faviconURL ?? "") {
+            guard let recentlyClosedTabs = NSUserDefaults.standardUserDefaults().objectForKey("recentlyClosedTabs") else {
+                var recentlyClosedTabsArray = [ClosedTab]()
+                recentlyClosedTabsArray.append(recentlyClosedTab)
+                let archivedTabsArray = NSKeyedArchiver.archivedDataWithRootObject(recentlyClosedTabsArray)
+                NSUserDefaults.standardUserDefaults().setObject(archivedTabsArray, forKey: "recentlyClosedTabs")
+                return
+            }
+            var unarchivedTabsArray = NSKeyedUnarchiver.unarchiveObjectWithData(recentlyClosedTabs as! NSData) as! [ClosedTab]
+            unarchivedTabsArray.append(recentlyClosedTab)
+
+            let archivedTabsArray = NSKeyedArchiver.archivedDataWithRootObject(unarchivedTabsArray)
+            NSUserDefaults.standardUserDefaults().setObject(archivedTabsArray, forKey: "recentlyClosedTabs")
+        }
+    }
+
+    public func getTabs() -> [ClosedTab] {
```

Comment:
Make this a `lazy` variable called `tabs`. The first time we access it, it will pull the tabs from prefs. Subsequent accesses can just return the array directly.